Tests/Packet++Test/Tests/InfiniBandTests.cpp

```diff
@@ -0,0 +1,224 @@
+#include "../TestDefinition.h"
+#include "../Utils/TestUtils.h"
+#include <sstream>
+#include "EndianPortable.h"
+#include "Logger.h"
+#include "MacAddress.h"
+#include "Packet.h"
+#include "EthLayer.h"
+#include "VlanLayer.h"
+#include "IPv4Layer.h"
+#include "IPv6Layer.h"
+#include "UdpLayer.h"
+#include "InfiniBandLayer.h"
+#include "PayloadLayer.h"
+#include "SystemUtils.h"
+
+std::string getProtocolTypeAsString(pcpp::ProtocolType protocolType)
+{
+	switch (protocolType)
+	{
+	case pcpp::Ethernet:
+		return "Ethernet";
+	case pcpp::VLAN:
+		return "Vlan";
+	case pcpp::IPv4:
+		return "IPv4";
+	case pcpp::UDP:
+		return "UDP";
+	case pcpp::TCP:
+		return "TCP";
+	case pcpp::Infiniband:
+		return "Infiniband";
+	case pcpp::HTTPRequest:
+	case pcpp::HTTPResponse:
+		return "HTTP";
+	case pcpp::GenericPayload:
+		return "Payload";
+	default:
+		return "Unknown";
+	}
+}
+
+PTF_TEST_CASE(InfiniBandPacketParsing)
+{
+	timeval time;
+	gettimeofday(&time, nullptr);
+
+	/*
+	uint8_t buffer1[] = {
+	    0x30, 0x46, 0x9a, 0x23, 0xfb, 0xfa, 0x6c, 0xf0,
+	    0x49, 0xb2, 0xde, 0x6e, 0x08, 0x00, 0x45, 0x00,
+	    0x00, 0x3c, 0x1a, 0x57, 0x00, 0x00, 0x80, 0x01,
+	    0x14, 0x65, 0x0a, 0x00, 0x00, 0x04, 0x01, 0x01,
+	    0x01, 0x01, 0x08, 0x00, 0x4d, 0x5a, 0x00, 0x01,
+	    0x00, 0x01, 0x61, 0x62, 0x63, 0x64, 0x65, 0x66,
+	    0x67, 0x68, 0x69, 0x6a, 0x6b, 0x6c, 0x6d, 0x6e,
+	    0x6f, 0x70, 0x71, 0x72, 0x73, 0x74, 0x75, 0x76,
+	    0x77, 0x61, 0x62, 0x63, 0x64, 0x65, 0x66, 0x67,
+	    0x68, 0x69,
+	};
+	*/
+
+	
+	// RRoce
+	uint8_t buffer1[] = {
```

Comment:
We usually add a `.dat` file and a `.pcap` file with relevant packets. You can see examples under `Tests/Packet++Test/PacketExamples`

Reply:
I have add InfinibandPacket.dat for the test example.